src/meta-srv/src/procedure/region_failover/update_metadata.rs

```diff
@@ -12,75 +12,166 @@
             catalog_name: failed_region.catalog.clone(),
             schema_name: failed_region.schema.clone(),
             table_name: failed_region.table.clone(),
         };
-        let mut global_value =
-            table_routes::get_table_global_value(&ctx.selector_ctx.kv_store, &global_key)
-                .await?
-                .with_context(|| TableNotFoundSnafu {
-                    name: common_catalog::format_full_table_name(
-                        &global_key.catalog_name,
-                        &global_key.schema_name,
-                        &global_key.table_name,
-                    ),
-                })?;
-
-        if let Some(mut region_numbers) = global_value
-            .regions_id_map
-            .remove(&failed_region.datanode_id)
-        {
+        let mut value = table_routes::get_table_global_value(&ctx.selector_ctx.kv_store, &key)
+            .await?
+            .with_context(|| TableNotFoundSnafu {
+                name: common_catalog::format_full_table_name(
+                    &key.catalog_name,
+                    &key.schema_name,
+                    &key.table_name,
+                ),
+            })?;
+
+        if let Some(mut region_numbers) = value.regions_id_map.remove(&failed_region.datanode_id) {
             region_numbers.retain(|x| *x != failed_region.region_number);
 
             if !region_numbers.is_empty() {
-                global_value
+                value
                     .regions_id_map
                     .insert(failed_region.datanode_id, region_numbers);
             }
         }
 
-        let region_numbers = global_value
+        let region_numbers = value
             .regions_id_map
-            .entry(self.candidate)
+            .entry(self.candidate.id)
             .or_insert_with(Vec::new);
         region_numbers.push(failed_region.region_number);
 
-        table_routes::put_table_global_value(
-            &ctx.selector_ctx.kv_store,
-            &global_key,
-            &global_value,
-        )
-        .await?;
+        table_routes::put_table_global_value(&ctx.selector_ctx.kv_store, &key, &value).await?;
+        info!(
+            "Region mappings in table global value (key = '{key}') are updated to {:?}. \
+            Failed region {} was on Datanode {}.",
+            value.regions_id_map, failed_region.region_number, failed_region.datanode_id,
+        );
         Ok(())
     }
+
+    async fn update_table_route(
+        &self,
+        ctx: &RegionFailoverContext,
+        failed_region: &RegionIdent,
+    ) -> Result<()> {
+        let table_name = TableName {
+            catalog_name: failed_region.catalog.clone(),
+            schema_name: failed_region.schema.clone(),
+            table_name: failed_region.table.clone(),
+        };
+        let key = TableRouteKey::with_table_name(failed_region.table_id as _, &table_name);
+        let value = table_routes::get_table_route_value(&ctx.selector_ctx.kv_store, &key).await?;
```

Comment:
I feel like things have been complicated, so here is my simplified idea:
1. add candidate peer to the tail of `TableRouteValue.peers`, return inserted index , (no need to consider duplication)
2. Iter the `TableRouteValue.region_routes` to find the failed region, then set the `leader_peer_index` as candidate's index

Reply:
We can move the code of `meta_client::rpc::TableRoute` to the common-meta module, so that we no longer need to depend on meta_client.